repos/system_upgrade/common/actors/commonleappdracutmodules/files/dracut/85sys-upgrade-redhat/do-upgrade.sh

```diff
@@ -338,6 +346,14 @@ mount -o "remount,rw" "$NEWROOT"
 
 ##### do the upgrade #######
 (
+    # check if leapp previously failed in the initramfs, if it did return to the emergency shell
+    [ -f "$NEWROOT$LEAPP_FAILED_FLAG_FILE" ] && {
+        echo "Found file $NEWROOT$LEAPP_FAILED_FLAG_FILE"
+        echo "Leapp previously failed and cannot continue, returning back to emergency shell"
```

Comment:
Sorry, one more thing. I guess we should advice users to open a BZ and attaching $NEWROOT/var/log/leapp/leapp-upgrade.log (or journal) as they are already messing around in the intiramdisk.
@pirat89 ^^^ wdyt?
I am going to test it again in the meantime.